### PR TITLE
Preferred and regular times availability feature for both weekly and specific events

### PR DESCRIPTION
**Pull request by ZhenyuYu1**
New Features:
1. Users can change their time preferences between regular and preferred using a toggle.
2. Regular times are green and preferred times are blue in the availability grid.
3. Preferred times are listed in descending order of number of participants on the side of the view-event page.

Create-event page using regular and preferred times.
Screenshot 2024-10-29 at 12 38 19 AM

Editing availability with both regular and preferred using toggle.
Screenshot 2024-10-29 at 12 46 25 AM

View-event page showing preferred times listed on the side.
Screenshot 2024-10-29 at 12 43 09 AM

View-event page with multiple users selecting preferred times.
Screenshot 2024-10-29 at 12 44 51 AM

Updated FAQs about preferred times.
Screenshot 2024-11-18 at 12 52 41 PM

### CODE REVIEW

File: src/components/AvailabilityGrid.tsx [hunk 1]
```diff
@@ -287,150 +302,222 @@ const Grid = ({
...
     const shade = shades[index]
+
+    // Check if the time segment is in the schedule
+    // const segments = schedule[date] || []
+    // for (const segment of segments) {
+    //   if (time === segment.beginning) {
+    //     console.log('TRUE')
+    //   }
+    // }
+
```
Comment (pselvana) [inline src/components/AvailabilityGrid.tsx]: Needed?

File: src/components/AvailabilityGrid.tsx [hunk 2]
```diff
@@ -287,150 +302,222 @@ const Grid = ({
...
-                className="flex items-center justify-center border-gray-300 text-sm text-gray-600"
-                style={{ height: '2rem' }}
-              ></div>
+            {timeArray.map(
+              (time, index) =>
+                index !== timeArray.length && (
+                  <div
+                    key={index}
+                    className="flex items-start justify-start border-gray-300 pr-2 text-xs text-gray-600"
+                    style={{ height: '64px', lineHeight: '64px' }}
```
Comment (pselvana) [inline src/components/AvailabilityGrid.tsx:372]: any specific reason for this styling? just curious.

**Review by pselvana — APPROVED** [review]
Can you update the FAQ and the instructions to discuss preferred times please? It's a good differentiator from all other sites so good to highlight and explain.  I'll approve but please make that change as well.

1 nitpick and 1 question on the review but nothing major requiring rework.

File: src/components/AvailabilityGrid.tsx (hunk 2 above)
Comment (ZhenyuYu1) [reply src/components/AvailabilityGrid.tsx:372]: I think the styling is the same as it was before.

File: src/components/AvailabilityGrid.tsx (hunk 1 above)
Comment (ZhenyuYu1) [reply src/components/AvailabilityGrid.tsx]: Removed all console logs.

**Pull request merged**